dask/dataframe/io.py

```diff
@@ -724,10 +738,47 @@ def one_path_one_key(path, key, start, stop, columns, chunksize, lock):
     if (start != 0 or stop is not None) and len(keys) > 1:
         raise NotImplementedError(read_hdf_error_msg)
     from .multi import concat
-    return concat([one_path_one_key(path, k, start, s, columns, chunksize, lock)
+    return concat([one_path_one_key(path, k, start, s, columns, chunksize,
+                                    lock, division_column)
                    for k, s in zip(keys, stops)])
 
 
+def _divide_buffer(buffer, division_column, start, stop, chunksize):
+    divisions = list()
+    starts = [start]
+    stops=list()
+    if isinstance(division_column, string_types):
+        _div_col = lambda x: x[division_column]
+    elif division_column:
+        _div_col = lambda x: pd.Series(x.index)
+    for i, chunk in enumerate(buffer):
```

Comment:
Also, this `_divide_buffer` should only be run when `read_hdf`is run. It returns starts, stops, and divisions. That is used to construct the dask dictionaries. So if I'm thinking about this right, this should never run during compute statements, its only used to set up the correct places for dask to start and stop and get clean divisions, even with repeating values in the division colum (provided that division is sorted.)